src/components/vm/vmReplaceSpiceDialog.jsx

```diff
@@ -30,42 +36,98 @@ import { fmt_to_fragments } from 'utils.jsx';
+    const isMultiple = spiceVMs?.length > 1;
+
     const Dialogs = useDialogs();
     const [error, dialogErrorSet] = useState(null);
     const [inProgress, setInProgress] = useState(false);
+    const defaultSelected = isMultiple ? [selectionId(spiceVMs[0])] : null;
+    const [selected, setSelected] = useState(defaultSelected);
 
-    function onReplace() {
+    async function onReplace() {
         setInProgress(true);
 
-        return domainReplaceSpice({ connectionName, id: vmId })
-                .then(() => Dialogs.close())
-                .catch(exc => {
-                    // we don't know of any case where this would fail, so this is all belt-and-suspenders
-                    // still, give the user some hint how to go on
-                    console.warn("Failed to replace SPICE devices:", exc);
-                    dialogErrorSet({
-                        dialogError: _("Failed to replace SPICE devices"),
-                        dialogErrorDetail: fmt_to_fragments(
-                            _("Please see $0 how to reconfigure your VM manually."),
-                            <a href="https://access.redhat.com/solutions/6955095" target="_blank" rel="noopener noreferrer">
-                                https://access.redhat.com/solutions/6955095
-                            </a>),
-                    });
-                })
-                .finally(() => setInProgress(false));
+        try {
+            if (isMultiple) {
+                // convert them serially, to avoid hammering libvirt with too many parallel requests
+                for (const sel of spiceVMs)
+                    if (selected.includes(selectionId(sel)))
+                        await domainReplaceSpice({ connectionName: sel.connectionName, id: sel.id });
+            } else {
+                await domainReplaceSpice({ connectionName: vm.connectionName, id: vm.id });
+            }
+            Dialogs.close();
+        } catch (ex) {
+            // we don't know of any case where this would fail, so this is all belt-and-suspenders
+            // still, give the user some hint how to go on
+            console.warn("Failed to replace SPICE devices:", ex);
+            setInProgress(false);
+            dialogErrorSet({
+                dialogError: _("Failed to replace SPICE devices"),
+                dialogErrorDetail: fmt_to_fragments(
+                    _("Please see $0 how to reconfigure your VM manually."),
+                    <a href="https://access.redhat.com/solutions/6955095" target="_blank" rel="noopener noreferrer">
+                        https://access.redhat.com/solutions/6955095
+                    </a>),
+            });
+        }
+    }
+
+    let vmSelect = null;
+
+    if (isMultiple) {
+        const onSelect = (_ev, item) => setSelected(
+            selected.includes(item)
+                ? selected.filter(id => id !== item)
+                : [...selected, item]
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1416-20240209-140217-7035aafd-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)